core/events.go

```diff
@@ -27,7 +27,10 @@ type NewTxsEvent struct{ Txs []*types.Transaction }
 // ReannoTxsEvent is posted when a batch of local pending transactions exceed a specified duration.
 type ReannoTxsEvent struct{ Txs []*types.Transaction }
 
-// NewMinedBlockEvent is posted when a block has been imported.
+// NewSealedBlockEvent is posted when a block has been sealed.
```

Comment:
Why add a new `NewSealedBlockEvent`, it seems the same usage as `NewMinedBlockEvent`.

Reply:
> Why add a new `NewSealedBlockEvent`, it seems the same usage as `NewMinedBlockEvent`.

before writing the block into db, we can only broadcast blocks directly
after writing it into db, we can broadcast the block hash

so need a way to distinguish these tow scenarios

Reply:
> > Why add a new `NewSealedBlockEvent`, it seems the same usage as `NewMinedBlockEvent`.
> 
> before writing the block into db, we can only broadcast blocks directly after writing it into db, we can broadcast the block hash
> 
> so need a way to distinguish these tow scenarios

`NewMinedBlockEvent` is for miner, `NewSealedBlockEvent` is for both miner and import, right?

Reply:
noted, for `NewSealedBlockEvent` if we broadcast the block hash, the remote peers could request the block, we may not able to response as it has not been written into db.